debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectInfrastructure.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.testing.testcontainers;
+
+
+/**
+ * Connect Testcontainers infrastructure handling.
+ */
+
+public class ConnectInfrastructure {
```

Comment:
I don't think we'll need this. As when someone wants to populate a Kafka Connect instance with the DebeziumContainer (which is a Kafka Connect container), the setup needs to be done outside anyway to inject related connector code etc.

Reply:
Taking it a step further, why do we need *any* of those *Infrastructure classes? They seem to add mostly complexity, for no real benefit. `PostgresInfrastructure` in particular has quite a few methods which don't seem to be used at all even. The Testcontainers API is simple and easy enough, tests can just interact with it, we don't need (and should not have) another abstraction on top of it.

Re `ApicurioConnectContainer`, I don't think there's need for that either; we should just have a method `enableApicurioConverters()` on the existing `DebeziumContainer` which sets the right environment variable for enabling those converters (see description of the Debezium image in our docker-containers repo, and the [Apicurio Compose file](https://github.com/debezium/debezium-examples/blob/master/tutorial/docker-compose-mysql-apicurio.yaml#L43) of the tutorial example).

We could have an `ApicurioRegistryContainer` though, which e.g. provides that `waitingFor()` logic OOTB.

Reply:
The `*Interface` class was meant for cases where we don't need any custom `GenericContainer<?>` extension (like the DebeziumContainer) and to avoid dependency to Testcontainers internal API. It just provided slightly customized default images using the external API.
But it's valid we could also create a new `*Container` class instead. But then we are architecturally bound to Testcontainers' API changes.

Reply:
Yes, I'm not concerned about changes to the `GenericContainer` API.